shortfin/python/shortfin_apps/llm/components/kvcache/trie_attention_cache.py

```diff
@@ -135,9 +136,13 @@ def __init__(self, page_pool: PagePool, tokens_per_page: int):
         self.root = TrieNode(tokens=tuple(), page=dummy_page)
         self.leaves: Set[TrieNode] = set()
         self._lock: Lock = Lock()
-        self._allocated_pages: List[PageInfo] = []
+        self._duplicated_pages: List[
+            PageInfo
+        ] = (
+            []
+        )  # pages that are duplicated from existing pages in Trie tree. These pages can be safely freed when calling release_pages.
 
-    def fork_pages(self, pages: List[PageInfo], tokens: list[int]) -> TrieCacheInfo:
+    def fork_pages(self, tokens: list[int], cache_info: TrieCacheInfo) -> TrieCacheInfo:
```

Comment:
Doesn't have to be this PR but can we potentially move away from `fork_pages`? I feel like this could be more easily handled at individual cases.